hnn_core/dipole.py

```diff
@@ -17,6 +18,43 @@ def _hammfilt(x, winsz):
     return convolve(x, win, 'same')
 
 
+# Savitzky-Golay filtering, lifted and adapted from mne-python (0.22)
+def _savgol_filter(data, h_freq, sfreq):
+    """Filter the data using Savitzky-Golay polynomial method.
+
+    Parameters
+    ----------
+    data : array-like
+        The data to filter (1D)
+    h_freq : float
+        Approximate high cutoff frequency in Hz. Note that this
+        is not an exact cutoff, since Savitzky-Golay filtering
+        is done using polynomial fits
+        instead of FIR/IIR filtering. This parameter is thus used to
+        determine the length of the window over which a 5th-order
+        polynomial smoothing is applied.
+    sfreq : float
+        Sampling rate
+
+    Returns
+    -------
+    filt_data : array-like
+        The filtered data
+    """  # noqa: E501
+    from scipy.signal import savgol_filter
+
+    h_freq = float(h_freq)
+    if h_freq >= sfreq / 2.:
+        raise ValueError('h_freq must be less than half the sample rate')
+
+    # savitzky-golay filtering
+    window_length = (int(np.round(sfreq / h_freq)) // 2) * 2 + 1
+    # loop over 'agg', 'L2', and 'L5'
+    filt_data = savgol_filter(data, axis=-1, polyorder=5,
```

Comment:
Worth exposing `polyorder` to the user? Or is 5th order pretty standard for this method

Reply:
We could, but leave out for simplicity. I guess Eric decided on that in mne-python. This is a purely viz-function, any quantitative work should apply more formal filters.